tests/store/artifact/test_models_artifact_repo.py

```diff
@@ -111,50 +111,99 @@ def test_models_artifact_repo_init_with_stage_uri_and_not_using_databricks_regis
+        models_repo.repo = mock.Mock(**{"download_artifacts.return_value": str(dummy_file)})
+
+        models_repo.download_artifacts("artifact_path", str(tmp_path))
 
+        models_repo.repo.download_artifacts.assert_called_once_with("artifact_path", str(tmp_path))
 
-def test_models_artifact_repo_add_registered_model_meta_file():
-    from mlflow.store.artifact.models_artifact_repo import REGISTERED_MODEL_META_FILE_NAME
 
+@pytest.mark.skipif(is_windows(), reason="This test fails on Windows")
+def test_models_artifact_repo_add_registered_model_meta_file(tmp_path):
     artifact_path = "artifact_path"
-    dst_path = "dst_path"
+    model_name = "MyModel"
+    model_version = "12"
+    artifact_location = str(tmp_path.joinpath(artifact_path))
+
+    # Create a directory and MLmodel file in tmp_path. These are the two conditions required
+    # to initiate the creation of the registered_model_meta_file
+    model_dir = tmp_path / "model_dir"
+    model_dir.mkdir()
+    mlmodel_file = model_dir / "MLmodel"
+    mlmodel_file.write_text("dummy content")
+
+    with mock.patch.object(
+        MlflowClient, "get_model_version_download_uri", return_value=artifact_location
+    ), mock.patch.object(
+        ModelsArtifactRepository, "_add_registered_model_meta_file"
+    ) as add_meta_mock:
+        models_repo = ModelsArtifactRepository(f"models:/{model_name}/{model_version}")
+        models_repo.repo = mock.Mock(**{"download_artifacts.return_value": str(model_dir)})
+
+        models_repo.download_artifacts(artifact_location, str(tmp_path))
+
+        add_meta_mock.assert_called_once()
+
+
+def test_models_artifact_repo_does_not_add_meta_for_file(tmp_path):
+    artifact_path = "artifact_file.txt"
+    model_name = "MyModel"
+    model_version = "12"
     artifact_location = f"s3://blah_bucket/{artifact_path}"
-    artifact_dst_path = f"{dst_path}/{artifact_path}"
+
+    dummy_file = tmp_path / artifact_path
+    dummy_file.write_text("dummy content")
+
+    with mock.patch.object(
+        MlflowClient, "get_model_version_download_uri", return_value=artifact_location
+    ), mock.patch.object(
+        ModelsArtifactRepository, "_add_registered_model_meta_file"
+    ) as add_meta_mock:
+        models_repo = ModelsArtifactRepository(f"models:/{model_name}/{model_version}")
+        models_repo.repo = mock.Mock(**{"download_artifacts.return_value": str(dummy_file)})
+
+        models_repo.download_artifacts(artifact_path, str(tmp_path))
+
+        add_meta_mock.assert_not_called()
+
+
+def test_models_artifact_repo_does_not_add_meta_for_directory_without_mlmodel(tmp_path):
+    artifact_path = "artifact_directory"
     model_name = "MyModel"
     model_version = "12"
+    artifact_location = f"s3://blah_bucket/{artifact_path}"
+
+    # Create a directory without an MLmodel file
+    dummy_dir = tmp_path / artifact_path
+    dummy_dir.mkdir()
+    dummy_file = dummy_dir / "dummy_file.txt"
+    dummy_file.write_text("dummy content")
```

Comment:
```suggestion
    dummy_file.touch()
```

We can use `touch` if the content is not used.